repos/c2sm/packages/icon/package.py

```diff
@@ -708,22 +715,34 @@ def configure_directory(self):
 
         Overides function from spack.build_systems.autotools
 
-        CAUTION: Does only work if Spack is inside the git-repo
-                 of ICON, otherwise "git rev-pars --show-toplevel"
-                 fails!
-
         """
 
-        Git = which('git', required=True)
-        git_root = Git('rev-parse', '--show-toplevel',
-                       output=str).replace("\n", "")
-        if git_root != self.stage.source_path:
-            # mark out-of-source build for function
-            # copy_runscript_related_input_files
-            self.out_of_source_build = True
-            return git_root
-        else:
-            return self.stage.source_path
+        source_path = self.build_directory
+
+        # dev_path is indicator for dev-build or develop
+        # only case when out-of-source build are possible
+        if "dev_path" in self.spec.variants:
+            Git = which('git', required=True)
+            git_root = Git('rev-parse',
+                           '--show-toplevel',
+                           output=str,
+                           fail_on_error=True).replace("\n", "")
+            if git_root != source_path:
+                # mark out-of-source build for function
+                # copy_runscript_related_input_files
+                self.out_of_source_build = True
+                self.out_of_source_configure_directory = git_root
+                return git_root
+
+        return source_path
+
+    @run_before('configure')
+    def report_out_of_source_directories(self):
+        if self.out_of_source_build:
+            tty.warn(f'build-directory: {self.build_directory}')
+            tty.warn(
+                f'configure-directory: {self.out_of_source_configure_directory}'
+            )
```

Comment:
This is more of an information than a warning.

Reply:
Switch to `info`